pslab/external/motor.py

```diff
@@ -70,3 +72,27 @@ def _get_duty_cycle(self, angle):
         angle *= self._max_angle_pulse - self._min_angle_pulse  # Scale
         angle += self._min_angle_pulse  # Offset
         return angle / (self._frequency**-1 * MICROSECONDS)
+
+
+class RoboticArm:
+    """Robotic arm controller for up to 4 servos."""
+
+    MAX_SERVOS = 4
+
+    def __init__(self, servos: List[Servo]) -> None:
+        if len(servos) > RoboticArm.MAX_SERVOS:
+            raise ValueError(
+                f"At most {RoboticArm.MAX_SERVOS} servos can be used, got {len(servos)}"
+            )
+        self.servos = servos
+
+    def run_schedule(self, timeline: List[List[int]], time_step: float = 1.0) -> None:
+        """Move servos based on a time-based angle schedule."""
+        tl_len = len(timeline[0])
+        if not all(len(tl) == tl_len for tl in timeline):
+            raise ValueError("All timeline entries must have the same length")
```

Comment:
Also, check that the number of timelines (`len(timeline)`) is equal to the number of servos (`len(self.servos)`).